ipn/local.go

```diff
@@ -366,7 +368,7 @@ func (b *LocalBackend) Start(opts Options) error {
 		Persist:         *persist,
 		ServerURL:       b.serverURL,
 		AuthKey:         opts.AuthKey,
-		Hostinfo:        hostinfo,
+		Hostinfo:        hostinfo.Clone(),
```

Comment:
Is this even safe to clone (which involves reading its fields) without the lock held?

Reply:
This is not-quite-obvious because it relies on the fact that nothing else is accessing hostinfo concurrently yet (part of why I ended up moving `readPoller` out of `NewLocalBackend`). This is also the reason for the comment below. That said, this `Clone` is actually superfluous, as I now notice that `controlclient.New` clones it again. We could just keep `b` locked, of course, to be future-proof.

Reply:
Hm, unfortunately, my `-race -count=10000` test just failed due to some very rate timeout (`-race -count=1000` succeeded several times). Let me look at that.

Reply:
Never mind, I guess the timeout was just due to the high count. Didn't expect it to actually go over 10 minutes.